Marlin/src/HAL/LINUX/inc/Conditionals_adv.h

```diff
@@ -20,3 +20,8 @@
  *
  */
 #pragma once
+
+// Redirect strlcpy to hal.str_lcpy
+#ifndef strlcpy
```

Comment:
Did you have cases that made this `#ifdef` necessary? I'm not sure if it works on function names, or only other macros. It might just always evaluate to false?

Reply:
Yes that works.
There are other examples in Marlin/src/HAL/NATIVE_SIM/inc/Conditionals_adv.h

And if you test it with

```
#define strlcpy(dest, source, totalsize) hal.str_lcpy(dest, source, totalsize)
#ifndef strlcpy
   #error 
#endif
```

The #error is not executed

Reply:
A `#define strlcpy blah` will cause any use of `strlcpy` in code that follows the `#define` to be replaced by `blah` so it effectively diverts all calls to `strlcpy` to `blah`, even if a proper `strlcpy` function exists. This is one way to get around the fact that `#ifndef` cannot see functions but only preprocessor macros. This is fine for a platform like Linux where we know there really isn't any `strlcpy` (also, where it matters less) but we wouldn't want to apply it to platforms with a real `strlcpy`.